doc/developer-guide/internal-libraries/Extendible.en.rst

```diff
@@ -0,0 +1,135 @@
+   auto fld_food_weight = Food.schema.addField<STATIC,float>("weight");
+   auto fld_expr_date = Food.schema.addField<STATIC,time_t>("expire date");
+   Food banana;
+
+   class Car : Extendible<Car> {}
+   auto fld_max_speed = Car.schema.addField<STATIC,float>("max_speed");
+   auto fld_car_weight = Car.schema.addField<STATIC,float>("weight"); // 'weight' is unique within 'Car'
+   Car camry;
+
+   // Common user errors
+
+   float expire_date = banana.get(fld_expr_date);
+   //^^^                                              Compile error: cannot convert time_t to float
+   float speed = banana.get(fld_max_speed);
+   //                       ^^^^^^^^^^^^^             Compile error: fld_max_speed is not of type Extendible<Food>::FieldId
+   float weight = camry.get(fld_food_weight);
+   //                       ^^^^^^^^^^^^^^^           Compile error: fld_food_weight is not of type Extendible<Car>::FieldId
+
+Field Access Types
+------------------
+.. _AccessType:
+
+============   =======================================   ============================================================   ========================================   ================================================
+Enums          Allocates                                 API                                                            Pros                                       Cons
+============   =======================================   ============================================================   ========================================   ================================================
+ATOMIC         std::atomic<FieldType>                    :code:`get(FieldId)`                                           Leverages std::atomic<> API. No Locking.   Only works on small data types.
+BIT            1 bit from packed bits                    :code:`get(FieldId), readBit(FieldId), writeBit(FieldId)`      Memory efficient.                          Cannot return reference.
+:ref:ACIDPTR   shared_ptr<FieldType> = new FieldType()   :code:`get(FieldId), writeAcidPtr(FieldId)`                    Avoid skew in non-atomic structures.       Non-contiguous memory allocations. Uses locking.
+STATIC         FieldType                                 :code:`get(FieldId), init(FieldId)`                            Const reference. Fast. Type safe.          No concurrency protection.
+DIRECT         FieldType                                 :code:`get(FieldId)`                                           Direct reference. Fast. Type safe.         No concurrency protection.
+C_API          a number of bytes                         :code:`get(FieldId_C)`                                         Can use in C.                              No concurrency protection. Not type safe.
+============   =======================================   ============================================================   ========================================   ================================================
```

Comment:
* Lots of strings here that are code but not marked up as such.
* Are `ATOMIC` and `ACIDPTR` also "type-safe" or no? Let's be consistent with pros/cons.